### PR TITLE
feat: internal audit logger

### LINKED ISSUE #1461: Audit log

### PR DESCRIPTION
**Pull request by TaduJR**
## What does this PR do?
This change tries to resolve #1461 by adding audit logger on v1 and v2 api and on nextjs backend.
These are the list of events handled on Teams and Organizations
```bash
Event-Types
	- Modified
		Event Setup
			Title
			Duration
			Location
		Assignment
		        Scheduling Type
	- Created
	- Delete

Bookings
	- Modified
		Reschedule
		Edit Location
	- Created
	- Cancelled
```

- Fixes #1461 (GitHub issue number)
- Fixes CAL-1461 (Linear issue number - should be visible at the bottom of the GitHub issue description)

https://www.loom.com/share/db03dbc9794d489bafcdace4a1eee664

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set? No
- What are the minimal test data to have? Team/Organization and User Entity Records
- What is expected (happy path) to have (input and output)? 
- Any other important info that could help to test that PR?

## Checklist

- I haven't commented my code, particularly in hard-to-understand areas
/claim #1461

### CODE REVIEW

**Comment by TaduJR** [comment]
Moving to this PR for more cleaner approach from https://github.com/calcom/cal.com/pull/15948

**Comment by anikdhabal** [comment]
@TaduJR really great work, could you pls fix the conflicts🙏

**Comment by TaduJR** [comment]
Thank you @anikdhabal. I thought the team abandoned this issue.
Ok, I will fix the conflicts.

**Comment by zomars** [comment]
> Thank you @anikdhabal. I thought the team abandoned this issue. Ok, I will fix the conflicts.

/tip 20

Thanks for the effort so far. We really appreciate it 🙏🏽

**Comment by TaduJR** [comment]
Thank you @zomars. Working on the conflicts.

**Comment by TaduJR** [comment]
Hello @zomars. I am sorry for not finishing it quickly. I was busy. I am working on the it right now to pass the tests. Sorry I should have converted it to draft.

**Comment by zomars** [comment]
> Hello @zomars. I am sorry for not finishing it quickly. I was busy. I am working on the it right now to pass the tests. Sorry I should have converted it to draft.

It's ok I'm just getting another round of review. Appreciate the patience 🙏🏽

**Review by zomars — CHANGES_REQUESTED** [review]
I'm seeing that you're still working on this. Could you check the test. And while at it, could you add some video demos?

File: turbo.json [hunk 1]
```diff
@@ -454,6 +454,7 @@
     "NEXT_PUBLIC_ORG_SELF_SERVE_ENABLED",
     "NEXT_PUBLIC_API_V2_ROOT_URL",
     "NEXT_PUBLIC_VAPID_PUBLIC_KEY",
-    "VAPID_PRIVATE_KEY"
+    "VAPID_PRIVATE_KEY",
+    "AUDIT_LOG_ENABLED"
```
[the remark this answers is not in the document]
Comment (TaduJR) [reply turbo.json]: Done.

**Comment by TaduJR** [comment]
> I'm seeing that you're still working on this. Could you check the test. And while at it, could you add some video demos?

Hello @zomars. Fixed the PR to pass the test also checkout the loom.

**Review by zomars — CHANGES_REQUESTED** [review]
Just finished reviewing the code and while the functionality is delivering as promised. The code feels way too intrusive to the core of the product itself. We strive to make the code legible and maintainable as possible. We're also trying to follow well tested and stablished conventions like Clean Code practices.

Will close this PR for now until we make more robust requirements on the desired approach code-wise.

Thank you again for your contribution 🙏

**Pull request closed without merging**